backend/drive/drive.go

```diff
@@ -2582,6 +2587,122 @@ func (f *Fs) Hashes() hash.Set {
 	return hash.Set(hash.MD5)
 }
 
+func (f *Fs) changeChunkSize(chunkSizeString string) (err error) {
+	chunkSizeInt, err := strconv.ParseInt(chunkSizeString, 10, 64)
+	if err != nil {
+		return errors.Wrap(err, "couldn't convert chunk size to int")
+	}
+	chunkSize := fs.SizeSuffix(chunkSizeInt)
+	if chunkSize == f.opt.ChunkSize {
+		return nil
+	}
+	err = checkUploadChunkSize(chunkSize)
+	if err == nil {
+		f.opt.ChunkSize = chunkSize
+	}
+	return
+}
+
+func (f *Fs) changeServiceAccountFile(file string) (err error) {
+	fs.Debugf(nil, "Changing Service Account File from %s to %s", file, f.opt.ServiceAccountFile)
+	if file == f.opt.ServiceAccountFile {
+		return nil
+	}
+	oldCredentials := f.opt.ServiceAccountCredentials
```

Comment:
I don't like the duplication of the rather trick initialization here...

I think it would be better if you factored that code out of the initialization first then just called that blob of code.

You can do your `undo` by making a copy of fs `fsCopy := *f` then using that to revert f.. (that might not work though because Fs has a mutex in..)

Reply:
Simplified the code a little bit. Let me know if that's fine.